lib/alephant/preview/server.rb

```diff
@@ -100,20 +107,41 @@ def render_preview
       end
 
       def render_component
-        view_mapper.generate(fixture_data)[template].render
+        view_mapper.generate(fixture_data)[template].render.tap do |content|
+          response["Content-Type"] = get_content_type(content)
+        end
       end
 
       def render_batch_component
+        content = render_component
+
         {
-          :component => template,
-          :options   => {},
-          :status    => 200,
-          :body      => render_component
+          :component    => template,
+          :options      => {},
+          :status       => 200,
+          :body         => content,
+          :content_type => get_content_type(content),
+          :sequence_id  => Time.now.to_i
         }
       end
 
       private
 
+      def get_content_type(content)
+        if is_json?(content)
+          "application/json"
+        else
+          "text/html"
+        end
+      end
+
+      def is_json?(content)
+        JSON.parse(content)
+        true
```

Comment:
You possibly don't need the explicit `true` here as if the JSON parses it'll return a "truthy" object, but it's cool to keep it in.